src/components/Dashboard.tsx

```diff
@@ -11,105 +11,84 @@ import {
   Th,
   Td,
   Tbody,
-  Link,
-  Checkbox,
+  Link, 
 } from "@chakra-ui/react";
-import {
-  OidcUserStatus,
-  useOidc,
-  useOidcAccessToken,
-  useOidcUser,
-} from "@axa-fr/react-oidc";
 import { Octokit } from "octokit";
 import { useDashboardController } from "../classes/DashboardController";
-//import { GitData } from "./GitData";
 import {useNavigate} from 'react-router-dom';
+import { table } from "console";
 
 interface TableItems {
   id: string;
   name: string;
   link: string;
+  login: string;
+  service:string;
 }
 
+
+
 export default function Dashboard() {
   const dashboardController = useDashboardController();
   const navigate = useNavigate();
-
   const [GitData, setGitData] = useState<Array<TableItems>>([]);
-  const [Tokens, setTokens] = useState<object>();
-
-  const [isCheckAll, setIsCheckAll] = useState<boolean>(false);
-  const [isCheck, setIsCheck] = useState<Array<string>>([]);
-  const [list, setList] = useState<Array<TableItems>>(GitData);
 
   useEffect(() => {
     const octokit = dashboardController.octokit;
-    octokit.request('GET /user/repos', {sort: 'pushed', direction: 'desc'}).then(({ data }) => {
-      setGitData(data.map(
-        eachRepo => ({id: eachRepo.id.toString(), name: eachRepo.name, link: eachRepo.html_url})
-        ));
-    });
-  }, [dashboardController]);
-  // const tokens = async () => {
-  //   const response = await fetch(serverDB + `/api/login`, requestOptions);
-  //   const json = await response.json();
-  //   //console.log(json);
-  //   console.log("Token fetched");
-  //   setTokens(json);
-  // };
+    //octokit.request('GET /user/repos', {sort: 'pushed', direction: 'desc'})
+    //.then(({ data }) => {
+    // setGitData(data.map(
+    //    eachRepo => ({id: eachRepo.id.toString(), name: eachRepo.name, link: eachRepo.html_url, service:eachRepo.service})
+    //    ));
+    //});
+
+    if (GitData.length==0){
+      octokit.request('GET /user')
+          .then( async ({data})=>{
+            const response = await fetch(`http://localhost:5001/api/config_repos/${data.login}`);
```

Comment:
would you please refactor this so that the list of configured repositories is stored in `ComputeService`, similar to how the list of computeServices are stored, along with a `useConfiguredRepositories` hook?

Reply:
Updated but have some blockers here